design/ErrorHandling.md

```diff
@@ -80,90 +79,244 @@ device.addEventListener('validationerror', (event) => {
 });
 ```
 
-## Object Creation
+## Object and Operation Status
+
+### Internal Nullability
 
 WebGPU objects are opaque handles.
 On creation, such a handle is "pending" until the backing object is created by the implementation.
 After that, a handle may refer to a successfully created object (called a "valid object"), or an error that occured during creation (called an "invalid object").
 
-When a WebGPU object handle is passed to an operation, the object will resolve (to "valid" or "invalid") before it is actually used by that operation.
+When a WebGPU object handle is passed to a WebGPU API call, the object must resolve (to "valid" or "invalid") before the implementation can perform validation on that call.
 
-### Error propagation of invalid objects
+### Categories of WebGPU Calls
 
-Using any invalid object in a WebGPU operation produces a validation error.
-The effect of an error depends on the type of a call:
+Each category of WebGPU call has its errors handled in a different way.
 
- - For object creation, the call produces a new invalid object.
- - For `GPUCommandEncoder` encoding methods, the `GPUCommandEncoder.finishEncoding` method will return an invalid object.
- - For other WebGPU calls, the call becomes a no-op.
+#### Initialization
 
-In each case, an error is logged to the error log.
+Creation of the adapter and device.
 
-## *Recovery*: Recoverable Errors
+  - `gpu.requestAdapter`
+  - `GPUAdapter.requestDevice`
 
-Recoverable errors are produced only by object creation.
-The status of an object can be retrieved asynchronously (see next section).
+Handled by "Fatal Errors" above.
 
-```
-enum GPUObjectStatus {
-    // The object is valid.
-    "valid",
-    // The object is invalid due to a non-fatal allocation failure.
-    // The application can use this as a signal to scale back resource usage, if possible.
-    "out-of-memory",
-    // The object is invalid for another, unrecoverable reason.
-    "invalid",
+#### Object-Returning
+
+WebGPU Object creation and getters.
+
+  - `GPUDevice.create*`
+  - `GPUTexture.create*View`
+  - `GPUCommandEncoder.finish`
+  - `GPUDevice.getQueue`
+  - `GPUSwapChain.getCurrentTexture`
+
+If there is an error, the returned object is invalid.
+
+See "Object Status" below.
+
+#### Object-Promise-Returning*
+
+Async WebGPU Object creation.
+
+  - `GPUDevice.createReady*`
```

Comment:
This PR obseletes `createReady` imho as an application could just wait for `getStatus(pipeline)`

Reply:
Good point, I thought this wouldn't work but I didn't think about it enough.